### PR TITLE
[Serverless] try to sanitize JSON payload received from Logs API before throwing an error

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

AWS sends an incorrect payload during the first Logs API call. Since the JSON is not well formatted, we can't parse it to build enhanced metrics.
This PR provides a temporary fix to allows the extension to parse properly the REPORT log.

### Motivation

Fix inconsistent enhanced metrics 

### Describe how to test your changes

Invoke the same function at least twice and check in the UI that all enhanced metrics are OK

log
3metricHits

### CODE REVIEW

File: pkg/serverless/aws/logs.go [hunk 1]
```diff
@@ -177,3 +179,24 @@ func createStringRecordForReportLog(l *LogMessage) string {
...
+		sanitizedData := removeInvalidTracingItem(data)
+		if err := json.Unmarshal(sanitizedData, &messages); err != nil {
+			return nil, errors.New("can't read log message")
+		}
+		return messages, nil
+	}
+	return messages, nil
+}
+
+// Temporary fix to handle malformed JSON tracing object
```
Comment (nhinsch) [inline pkg/serverless/aws/logs.go:198]: Nit: In Go, the comments above functions are supposed to always begin with the function name -- it's a documentation convention

So, `removeInvalidTracingItem is a temporary fix to...`

File: pkg/serverless/protocol.go [hunk 2]
```diff
@@ -289,32 +288,35 @@ func (l *LogsCollection) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-				logMessage := logConfig.NewChannelMessageFromLambda([]byte(message.StringRecord), message.Time, arn, lastRequestID, functionName)
-				l.ch <- logMessage
-			}
-		}
+		processLogsAPIPayload(l, messages)
 		w.WriteHeader(200)
 	}
 }
 
+func processLogsAPIPayload(l *LogsCollection, messages []aws.LogMessage) {
```
Comment (nhinsch) [inline pkg/serverless/protocol.go:301]: I would rename this to `processLogMessages` -- at this point it's not really the API payload any more, it has already been unmarshalled into the `[]aws.LogMessage` type.

**Review by nhinsch — APPROVED** [review]
Looks great, a couple of tiny things

**Comment by nhinsch** [comment]
I think you renamed a different function than the one I suggested 😂

File: pkg/serverless/aws/logs.go (hunk 1 above)
Comment (maxday) [reply pkg/serverless/aws/logs.go]: ✅

File: pkg/serverless/protocol.go (hunk 2 above)
Comment (maxday) [reply pkg/serverless/protocol.go]: ✅

**Pull request merged**